go-controller/pkg/crd/userdefinednetwork/v1/cudn.go

```diff
@@ -70,17 +76,22 @@ type NetworkSpec struct {
 	Localnet *LocalnetConfig `json:"localnet,omitempty"`
 
 	// Transport describes the transport technology for pod-to-pod traffic.
-	// Allowed values are "NoOverlay" and "Geneve".
+	// Allowed values are "NoOverlay", "Geneve", and "EVPN".
 	// - "NoOverlay": The network operates in no-overlay mode.
 	// - "Geneve": The network uses Geneve overlay.
+	// - "EVPN": The network uses EVPN transport.
 	// When omitted, the default behaviour is Geneve.
-	// +kubebuilder:validation:Enum=NoOverlay;Geneve
+	// +kubebuilder:validation:Enum=NoOverlay;Geneve;EVPN
 	// +optional
 	Transport TransportOption `json:"transport,omitempty"`
 	// NoOverlayOptions contains configuration for no-overlay mode.
 	// This is only allowed when Transport is "NoOverlay".
 	// +optional
 	NoOverlayOptions *NoOverlayOptions `json:"noOverlayOptions,omitempty"`
+	// EVPNConfiguration contains configuration for EVPN mode.
+	// This is only allowed when Transport is "EVPN".
+	// +optional
+	EVPNConfiguration *EVPNConfiguration `json:"evpnConfiguration,omitempty"`
```

Comment:
I find `evpnConfiguration`/`noOverlayOptions` as field names in our API incorrect but I might be nitpicky.
I think all of the fields are configuration/option and we are not adding the suffix in fields like:
`Layer2Config` -> `layer2`
`Layer3Config` -> `layer3` 
`TransportOption` -> `transport`

I think I would consider `noOverlay` and `evpn` a better choice that is consistent with existing fields. 
I know this is just applying what's in the enhancement but I would like to discuss it before it is too late.

Reply:
Yeah, I agree it makes sense to maintain consistency.

Reply:
this is done.